Duplicate linkage of oa_hash.h

Compiling on the dev branch now fails:

```
make[3]: Leaving directory '/usr/local/src/concord/src'
cc -shared -lcurl -o ../lib/libdiscord.so discord-refcount.o discord-rest.o discord-rest_request.o discord-rest_ratelimit.o discord-client.o discord-events.o discord-cache.o discord-loop.o discord-gateway.o discord-gateway_dispatch.o discord-messagecommands.o discord-timer.o discord-misc.o discord-worker.o application_command.o auto_moderation.o interaction.o audit_log.o channel.o emoji.o gateway.o guild.o guild_scheduled_event.o guild_template.o invite.o oauth2.o user.o voice.o webhook.o ../gencodecs/discord_codecs.o ../core/cog-utils.o ../core/concord-error.o ../core/concord-notifier.o ../core/concord-once.o ../core/io_poller.o ../core/user-agent.o ../core/websockets.o ../core/jsmn-find.o ../core/json-build.o ../core/logmod.o ../core/oa_hash.o ../core/priority_queue.o ../core/anomap.o ../core/threadpool.o ../core/queriec.o
/usr/bin/ld: ../core/oa_hash.o: in function `oa_hash_init':
oa_hash.c:(.text+0x0): multiple definition of `oa_hash_init'; ../core/jsmn-find.o:jsmn-find.c:(.text+0xa49): first defined here
/usr/bin/ld: ../core/oa_hash.o: in function `oa_hash_cleanup':
oa_hash.c:(.text+0x56): multiple definition of `oa_hash_cleanup'; ../core/jsmn-find.o:jsmn-find.c:(.text+0xa9f): first defined here
/usr/bin/ld: ../core/oa_hash.o: in function `oa_hash_get_entry':
oa_hash.c:(.text+0x11e): multiple definition of `oa_hash_get_entry'; ../core/jsmn-find.o:jsmn-find.c:(.text+0xb67): first defined here
/usr/bin/ld: ../core/oa_hash.o: in function `oa_hash_get':
oa_hash.c:(.text+0x211): multiple definition of `oa_hash_get'; ../core/jsmn-find.o:jsmn-find.c:(.text+0xc5a): first defined here
/usr/bin/ld: ../core/oa_hash.o: in function `oa_hash_set_entry':
oa_hash.c:(.text+0x258): multiple definition of `oa_hash_set_entry'; ../core/jsmn-find.o:jsmn-find.c:(.text+0xca1): first defined here
/usr/bin/ld: ../core/oa_hash.o: in function `oa_hash_set':
oa_hash.c:(.text+0x3f7): multiple definition of `oa_hash_set'; ../core/jsmn-find.o:jsmn-find.c:(.text+0xe40): first defined here
/usr/bin/ld: ../core/oa_hash.o: in function `oa_hash_remove':
oa_hash.c:(.text+0x443): multiple definition of `oa_hash_remove'; ../core/jsmn-find.o:jsmn-find.c:(.text+0xe8c): first defined here
/usr/bin/ld: ../core/oa_hash.o: in function `oa_hash_rehash':
oa_hash.c:(.text+0x556): multiple definition of `oa_hash_rehash'; ../core/jsmn-find.o:jsmn-find.c:(.text+0xf9f): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:84: ../lib/libdiscord.so] Error 1
make[2]: Leaving directory '/usr/local/src/concord/src'
make[1]: *** [Makefile:77: shared] Error 2
make[1]: Leaving directory '/usr/local/src/concord/src'
make: *** [Makefile:34: shared] Error 2
```

I suspect it probably has since `oa_hash` stuff was introduced into this project though I haven't confirmed that.

A preliminary search reveals that `oa_hash.h` is included in multiple files in this project:
```
# grep -R "oa_hash.h"
core/jsmn-find.h:#include "oa_hash.h" <------------------------------------
core/jsmn-find.h:#include "oa_hash.h"
core/oa_hash.c:#include "oa_hash.h" <------------------------------------

# grep -R "jsmn-find.h"
gencodecs/gencodecs.h:GENCODECS_PP_INCLUDE("jsmn-find.h")
gencodecs/discord_codecs.h:#include "jsmn-find.h"
core/jsmn-find.c:#include "jsmn-find.h" <------------------------------------
core/jsmn-find.h:#error "jsmn-find.h should be included after jsmn.h"
grep: .git/index: binary file matches
test/racecond.c:#include "jsmn-find.h"
include/discord-internal.h:#include "jsmn-find.h"
```

I don't see how this is valid - this will result in the code in `oa_hash.h` being linked in twice, resulting in the error above.

As a temporary workaround, running `echo "" > core/oa_hash.c` allows the library to compile successfully for now, but I'll let you implement the proper fix here as it's not clear how this was intended to be organized.

--- a/include/discord.h
+++ b/include/discord.h
@@ -210,6 +210,12 @@ void discord_shutdown_all(void);
 bool discord_shutdown_all_ongoing(void);
 
 /**
+ * @brief Backwards compatible alias for discord_shutdown_all()
+ * @deprecated since v3.0.0
+ */
+#define ccord_shutdown_async() discord_shutdown_all()
+
+/**
  * @brief Creates a Discord Client handle from a token
  * @see discord_get_logmod() to configure logging behavior
  *

